### PR TITLE
Update executorch pin

### PR DESCRIPTION
**Pull request by tugsbayasgalan**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #123764

### CODE REVIEW

**Review by larryliu0820 — APPROVED** [review]

**Review by ezyang — APPROVED** [review]

**Comment by cccclai** [comment]
Need to update the nightly as well, reference https://github.com/pytorch/executorch/pull/2944

File: .ci/docker/requirements-ci.txt [hunk 1]
```diff
@@ -295,13 +295,19 @@ tensorboard==2.13.0
...
 pywavelets==1.4.1 ; python_version < "3.12"
 pywavelets==1.5.0 ; python_version >= "3.12"
 #Description: This is a requirement of scikit-image, we need to pin
 # it here because 1.5.0 conflicts with numpy 1.21.2 used in CI
 #Pinned versions: 1.4.1
 #test that import:
 
+zstd
+#Description: This is a requirement for Executorch
+
```
Comment (kit1980) [inline .ci/docker/requirements-ci.txt:310]: I don't think these changes are needed.
From the log, the file that is used is from executorch https://github.com/pytorch/executorch/blob/main/.ci/docker/requirements-ci.txt

I'm not sure why the build is failing though.

**Comment by kit1980** [comment]
The build failures are because `pip install .` creates new pip env, and packages from the original env are not inherited.
Proper fix should be specifying the build time deps in pyproject.toml, like https://github.com/pytorch/executorch/pull/3110

Or use `python setup.py develop` instead of `pip install .`

**Pull request closed without merging**